File: Day30/Java/print_1_to_N_backtracking.java
public class print_1_to_N_backtracking {
    static void printval(int i,int x,int[] arr){
        if(i<1) return;
        printval(i-1, x, arr);
        arr[i-1] = i;
    }
    public static int[] printNos(int x) {
        int[] arr = new int[x];
        printval(x, x, arr);
        return arr;
    }
    public static void main(String[] args) {
        int n = 5;
        int[] ans = printNos(n);
        for(int i=0;i<ans.length;i++){
            System.out.print(ans[i]+" ");
        }
    }
}

File: Day30/Cpp/print1_to_N_backtracking.cpp
#include<bits/stdc++.h>
using namespace std;
void printval(int i,int x,vector<int> &y){
    if(i<1){
        return;
    }
    else{
        printval(i-1,x,y);
        y.push_back(i);
    }
}

vector<int> printNos(int x) {
    vector<int> v;
    printval(x,x,v);
    return v;
}




int main()
{
    int n;
    cin>>n;
    vector<int> v = printNos(n);
    for(auto i:v){
        cout<<i<<" ";
    }
    return 0;
}